src/Generating/FinishGen.cpp

```diff
@@ -2185,3 +2213,82 @@ void cFinishGenOrePockets::imprintSphere(
 
 
 
+cFinishGenForestRocks::cFinishGenForestRocks(int a_Seed, cIniFile & a_IniFile) : m_Noise(a_Seed)
+{
+}
+
+
+
+
+
+void cFinishGenForestRocks::GenFinish(cChunkDesc & a_ChunkDesc)
+{
+	auto Biome = a_ChunkDesc.GetBiome(cChunkDef::Width / 2, cChunkDef::Width / 2);
+	if ((Biome != biMegaTaiga) && (Biome != biMegaTaigaHills))
+	{
+		return;
+	}
```

Comment:
Should this come after we've picked an X, Z or is this intended?

Reply:
actually we could use the position x,z to check if that block is valid - you are right